src/diagram/units-manager.ts

```diff
@@ -0,0 +1,189 @@
+import { CreateUnitOptions, SymbolNode } from "mathjs";
+import { action, makeObservable, observable } from "mobx";
+import * as pluralize from "pluralize";
+import { IMathLib } from "./utils/mathjs-utils";
+
+export interface IUnit {
+  unit: string;
+  options?: CreateUnitOptions;
+}
+
+export class UnitsManager {
+  // A list of custom units added to mathjs. This list will grow as a user adds custom units.
+  // Built-in custom units can also be added here. They should look like:
+  // { unit: "cat", options: { aliases: ["cats"] } }
+  units: IUnit[] = observable.array<IUnit>();
+
+  constructor() {
+    makeObservable(this);
+  }
+
+  @action
+  addUnit(unit: string, options?: CreateUnitOptions) {
+    if (this.units.find((iunit: IUnit) => iunit.unit === unit)) return;
+    this.units.push({ unit, options });   
+  }
+
+  // This is not an action because we want changes in the units to be observed
+  // and trigger initializeMath to be run again
+  initializeMath(m: math.MathJsStatic) {
+    // The types don't give access to the Unit class object, but it is there.
+    const mathUnit = (m as any).Unit;
+
+    const isAlphaOriginal = mathUnit.isValidAlpha;
+    mathUnit.isValidAlpha = function (c: string): boolean {
+      return isAlphaOriginal(c) || c === "$";
+    };
+  
+    deleteUnits.forEach((u: string) => mathUnit.deleteUnit(u));
+    this.units.forEach((u: IUnit) => {
+      // createUnit has two versions:
+      //   createUnit(name: string, definition?: string | UnitDefinition, options?: CreateUnitOptions): Unit;
+      //   createUnit(units: Record<string, string | UnitDefinition>, options?: CreateUnitOptions): Unit;
+      // If u.options is undefined and we call `m.createUnit(u.unit, u.options);` Typescript's overloading
+      // support fails for some reason. So this is split out to 2 function calls.
+      if (u.options) {
+        m.createUnit(u.unit, u.options);
+      } else {
+        m.createUnit(u.unit);
+      }
+    });
+
+    return mathUnit;
+  }
+
+  // NOTE: this is both an action and a view. It is called by MST views to get the math.Unit
```

Comment:
I tried to update the comment to clarify this. Technically it is not a view, but it is used by real views, so that indirectly makes it function as a view.